packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -78,13 +78,13 @@ export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxStat
         <input
           id={ id }
           className={ classNames.field }
-          placeholder={ labelText }
+          placeholder={ labelText ? labelText : placeholder }
```

Comment:
Move this logic up ie `if (label) placeholder = label`, and this line can just be `placeholder={ placeholder }`  so your change in line 87 won't affect those who pass in `labelText`